### ISSUE TITLE
Import Watch Only Wallet generated by SeedHammer fails with `Error parsing UR CBOR`

### ISSUE THREAD

**Issue by Schnitzel** [sparrowwallet/sparrow#1329]
I got a weird error when trying to import the DescriptorQR generated by SeedHammer in Sparrow.

Steps to reproduce:

```
1. open sparrow
2. file > New Wallet
3. Give Wallet a name
4. click camera next to "Script Policy"
5. scan Descriptor QR code
6. error "Error parsing UR CBOR" arrives 
```
![photo_5168235088054431095_w](https://github.com/sparrowwallet/sparrow/assets/750999/20a179ac-c203-463d-9e27-da1b8645c965)

<details>

<summary>Sparrow Log files entries:</summary>

```
2024-03-17 22:42:46,628 ERROR [Thread-259] c.s.s.c.QRScanDialog [null:-1] Error parsing UR CBOR
java.lang.NullPointerException: Cannot read the array length because "<parameter1>" is null
	at com.sparrowwallet.drongo/com.sparrowwallet.drongo.Utils.bytesToHex(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.getKeyDerivation(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.getOutputDescriptor(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.extractResultFromUR(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.changed(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.changed(Unknown Source)
	at javafx.base@18/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at javafx.base@18/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base@18/javafx.beans.property.ObjectPropertyBase.fireValueChangedEvent(Unknown Source)
	at javafx.base@18/javafx.beans.property.ObjectPropertyBase.markInvalid(Unknown Source)
	at javafx.base@18/javafx.beans.property.ObjectPropertyBase.set(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.WebcamService.readQR(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.WebcamService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.WebcamService$1.call(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
2024-03-17 22:42:46,639 ERROR [JavaFX Application Thread] c.s.s.w.SettingsController [null:-1] Error scanning QR
com.sparrowwallet.sparrow.control.QRScanDialog$URException: Error parsing UR CBOR
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.extractResultFromUR(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.changed(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.changed(Unknown Source)
	at javafx.base@18/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at javafx.base@18/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base@18/javafx.beans.property.ObjectPropertyBase.fireValueChangedEvent(Unknown Source)
	at javafx.base@18/javafx.beans.property.ObjectPropertyBase.markInvalid(Unknown Source)
	at javafx.base@18/javafx.beans.property.ObjectPropertyBase.set(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.WebcamService.readQR(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.WebcamService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.WebcamService$1.call(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot read the array length because "<parameter1>" is null
	at com.sparrowwallet.drongo/com.sparrowwallet.drongo.Utils.bytesToHex(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.getKeyDerivation(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.control.QRScanDialog$QRResultListener.getOutputDescriptor(Unknown Source)
	... 19 common frames omitted
```

</details>

Importing the same DescriptorQR into Nunchuck on iOS works without a problem.

Sparrow Version: 1.8.4
macOS Sonoma 14.4 (23E214)

**Comment by Schnitzel** [comment]
btw, I know this is hard to debug without the actual data, I'm currently creating a test QR to reproduce the issues without any funds on it :)

**Comment by craigraw** [comment]
Thanks for the bug report - the UR Seedhammer produced was without a master fingerprint. Fixed in 9d0c35bc.
